blocks/extensions/saveload.py

```diff
@@ -56,6 +45,61 @@ def __init__(self, path, **kwargs):
 
     def do(self, callback_name, *args):
         """Pickle the main loop object to the disk."""
-        with open(self.path, "wb") as destination:
-            dill.dump(self.main_loop, destination, fmode=dill.CONTENTS_FMODE)
-        self.log_saving_done(self.path)
+        try:
+            self.main_loop.log.current_row[SAVING_DONE_TO] = self.path
+            with open(self.path, "wb") as destination:
+                dill.dump(self.main_loop, destination,
+                          fmode=dill.CONTENTS_FMODE)
+        except:
+            self.main_loop.log.current_row[SAVING_DONE_TO] = None
+
+
+class LoadFromDump(TrainingExtension):
+    """Loads a dump into the main loop.
+
+    Parameters
+    ----------
+    state_path : str
+        The path to the folder with dump.
+
+    """
+    def __init__(self, state_path, **kwargs):
+        super(LoadFromDump, self).__init__(**kwargs)
+        self.manager = MainLoopDumpManager(state_path)
+
+    def before_training(self):
+        if not os.path.exists(self.manager.folder):
+            logger.info("No dump found")
+            return
+        logger.info("Loading the state from {} into the main loop"
+                    .format(self.manager.folder))
+        try:
+            self.manager.load_to(self.main_loop)
+            self.main_loop.log.current_row[LOADED_FROM] = self.manager.folder
+        except:
+            logger.error("Failed to load the state:\n{}"
```

Comment:
Better to use `reraise_as("Failed to load state")` from `blocks.utils` here, that way you can still use PDB to do debugging.